Source/WebKit/Shared/Extensions/WebExtensionRegisteredScriptsSQLiteStore.cpp

```diff
@@ -0,0 +1,303 @@
+            completionHandler(scripts, errorMessage);
+        });
+    });
+}
+
+Vector<Ref<JSON::Object>> WebExtensionRegisteredScriptsSQLiteStore::getScriptsWithErrorMessage(String& errorMessage)
+{
+    assertIsCurrent(queue());
+
+    if (!openDatabaseIfNecessary(errorMessage, false))
+        return { };
+
+    ASSERT(errorMessage.isEmpty());
+
+    if (RefPtr rows = SQLiteDatabaseFetch(*database(), "SELECT * FROM registered_scripts"_s))
+        return getKeysAndValuesFromRowIterator(*rows);
+    return { };
+}
+
+Vector<Ref<JSON::Object>> WebExtensionRegisteredScriptsSQLiteStore::getKeysAndValuesFromRowIterator(Ref<WebExtensionSQLiteRowEnumerator> rows)
+{
+    Vector<Ref<JSON::Object>> results;
+
+    RefPtr row = rows->next();
+    while (row != nullptr) {
+        auto script = row->getString(1);
+
+        auto value = JSON::Value::parseJSON(script);
+        if (RefPtr object = value->asObject())
+            results.append(*object);
+        else
+            RELEASE_LOG_ERROR(Extensions, "Failed to deserialize registered content scripts for extension %s", uniqueIdentifier().utf8().data());
+
+        row = rows->next();
+    }
+
+    return results;
+}
+
+void WebExtensionRegisteredScriptsSQLiteStore::insertScript(const JSON::Object& script, Ref<WebExtensionSQLiteDatabase> database, String& errorMessage)
+{
+    assertIsCurrent(queue());
+
+    auto scriptID = script.getString(idKey);
+    ASSERT(!scriptID.isEmpty());
+
+    auto scriptData = script.toJSONString();
+    DatabaseResult result = SQLiteDatabaseExecute(database, "INSERT INTO registered_scripts (key, script) VALUES (?, ?)"_s, scriptID, scriptData);
+    if (result != SQLITE_DONE) {
+        RELEASE_LOG_ERROR(Extensions, "Failed to insert registered content script for extension %s.", uniqueIdentifier().utf8().data());
+        errorMessage = "Failed to add content script."_s;
+        return;
+    }
+}
+
+// MARK: Database Schema
+
+SchemaVersion WebExtensionRegisteredScriptsSQLiteStore::currentDatabaseSchemaVersion()
+{
+    return currentSchemaVersion;
+}
+
+DatabaseResult WebExtensionRegisteredScriptsSQLiteStore::createFreshDatabaseSchema()
+{
+    assertIsCurrent(queue());
+    ASSERT(database());
+
+    DatabaseResult result = SQLiteDatabaseExecute(*database(), "CREATE TABLE registered_scripts (key TEXT PRIMARY KEY NOT NULL, script BLOB NOT NULL)"_s);
+    if (result != SQLITE_DONE)
+        RELEASE_LOG_ERROR(Extensions, "Failed to create registered_scripts database for extension %s: %s (%d)", uniqueIdentifier().utf8().data(), lastErrorMessage().data(), result);
+    return result;
+}
+
+SchemaVersion WebExtensionRegisteredScriptsSQLiteStore::migrateToCurrentSchemaVersionIfNeeded()
+{
+    assertIsCurrent(queue());
+
+    auto currentDatabaseSchemaVersion = databaseSchemaVersion();
+    if (currentDatabaseSchemaVersion == 1 && currentSchemaVersion == 2) {
```

Comment:
Same, I think this needs to migrate any v1.

```suggestion
    if (currentDatabaseSchemaVersion == 1) {
```

Reply:
This still needs changed.